ID-6: JS-Doc implementation, components removal and Error Boundary inconsistency issue fixing

Sardor-M opened the pull request:
Those are the list of the changes fixed and being proposed: 

- [x] Rewrote JSDoc for all components and pages to follow consistent documentation practices
- [x] Removed unnecessary components and reorganized file structure for better code organization
- [x] Refactored routes and router component to properly utilize the Error Boundary component
- [x] Consolidated Error Boundary related components by moving them to a dedicated Error folder
- [x] Set ErrorFallback as the default component for Error Boundary
- [x] Removed all Firebase cloud functions, rules, and logs files

> Important to note: In this PR, i also added the linting issues fixes, so once you run the ``npm run lint``, you should get some linting issue regarding the kakaomap components, please refer to the comment in ID-6 section for clearance. 

- [x] Fixed the Lint Issues mentioned in the task: ID-5 

P.S: As far as i know, the white screen showing up instead of the app running is working as expected without any issues, once you run the app in local machine, let me know if there's any issue with that regard (happens again).

@iamalaziz 
I hope this all these commits has fixed the all the issues mentioned in the task: ID-6.

Thank you ~

iamalaziz commented on src/pages/Events/EventRegister/EventForm/EventForm.tsx:
```diff
@@ -0,0 +1,12 @@
+// type EventFormProps = {
```
*(minor)* As discussed in previous sessions, avoid leaving commented code in the codebase, especially as a way to temporarily disable functionality. It clutters th code, reduces readability, and makes it harder to distinguish between active and obsolete logic. If you need to experiment or temporarily remove code, you could use `git stash`. These tools are designed for safe, clean handling of in-progress changes without polluting the main codebase.

iamalaziz reviewed (changes requested):
Thank you for adding detailed `JSDoc` for functions. They are now much helpful and informative. 👍

iamalaziz commented:
Additionally, it would be best if you could standardize all imports.
* Using only alias imports. I found some full path imports. You can search with `../` to find them.
* Separate local imports from package imports with an empty line for better readibility.

Sardor-M commented on src/App.tsx:
```diff
@@ -1,23 +1,23 @@
+import { BrowserRouter } from 'react-router-dom';
+import RouterPage from '@/router/index';
+import ErrorBoundary from '@/components/Error/ErrorBoundary';
+import RootLayout from '@/pages/layout';
```
Ah right before the pushing the commit, i had changes the file name, so i think it has caused the import break, will be fixing it soon.

Sardor-M commented on src/components/Button/Button.tsx:
```diff
@@ -1,179 +1,184 @@
- * Renders a standard HTML button or a react-router-dom Link styled as a button,
- * based on the 'asLink' prop.
- *
  * @returns A button or Link element with specified styles.
```
Ah i see, will be fixing it, i forgot to double-check it before pushing it.

Sardor-M commented on src/components/Button/Button.tsx:
```diff
@@ -1,179 +1,184 @@
-      to = "/";
+    if (asLink) {
+        if (to === undefined) {
+            console.warn("Button: 'to' prop is required when 'asLink' is true.");
```
When adding the JSdoc to  Button Component, i should have checked the conflicting props issue,  thanks for mentioning ! Your suggestion union types clearly will be the best practice for our Button component case.

Soon will be fixed.

Sardor-M commented on src/components/Button/Button.tsx:
```diff
@@ -1,179 +1,184 @@
+                as={Link}
+                {...commonStyledProps}
+                {...linkProps} // ink-specific props pass qilamiz
+                {...rest} //  Remaining compatible props larni pass qilamiz
```
Yeap, i agree on this, will be only using the english in further cases, no worries !

Sardor-M commented on src/components/Button/Button.tsx:
```diff
@@ -1,179 +1,184 @@
-    variant,
-    size,
-    fullWidth,
+    variant = 'primary',
```
Nice catch, i replaced the Filter Button to use the Buttom Atomic Component, but i overwrite the some of the properties to get the desired styling we need.

Sardor-M commented on src/components/Header/HeaderDesktop.tsx:
```diff
@@ -1,201 +1,185 @@
-import { HeaderProps } from "@/types";
+import styled from 'styled-components';
+import { Link, NavLink as RouterNavLink } from 'react-router-dom';
+import kodjLogo from '@/static/assets/kodj_new.jpg';
```
Will be fixing those minor import issue.

Sardor-M commented:
> Additionally, it would be best if you could standardize all imports.
> 
> * Using only alias imports. I found some full path imports. You can search with `../` to find them.
> * Separate local imports from package imports with an empty line for better readibility.

I fixed all the imports except for the AuthContext component since i have yet touched on this component ( as we have planned to refactor the entire AuthContext logic in the next task, will also be  defaulted to use the type alias importing.)

Thanks for pointing out !

![Screenshot 2025-04-25 at 11 20 07 AM](https://github.com/user-attachments/assets/71845b08-f059-473e-b151-9dc6376cb71a)

Sardor-M commented:
> > Additionally, it would be best if you could standardize all imports.
> > 
> > * Using only alias imports. I found some full path imports. You can search with `../` to find them.
> > * Separate local imports from package imports with an empty line for better readibility.
> 
> I fixed all the imports except for the AuthContext component since i have yet touched on this component ( as we have planned to refactor the entire AuthContext logic in the next task, will also be defaulted to use the type alias importing.)
> 
> Thanks for pointing out !
> 
> ![Screenshot 2025-04-25 at 11 20 07 AM](https://private-user-images.githubusercontent.com/65296404/437266370-71845b08-f059-473e-b151-9dc6376cb71a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU1NDg4MTYsIm5iZiI6MTc0NTU0ODUxNiwicGF0aCI6Ii82NTI5NjQwNC80MzcyNjYzNzAtNzE4NDViMDgtZjA1OS00NzNlLWIxNTEtOWRjNjM3NmNiNzFhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI1VDAyMzUxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVjNzUxYmQ4YWI4ODM0NjFkNWJmODBiNDBjY2JiMTVmMzIzMjdkY2VhZjI4YjJmOTcyM2I2M2JlZmJlNTIzZDQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.0OQb2mOmGrxtZxKOB7c7qYO8qbqaBtYDbgZYa8BSX3w)

Also i removed the commented code components (which were based on tailwind styling)  and the duplicate i18n setting file.

commit ref: 4e1377b366fab808f0b329afda18fed32221ec98

Sardor-M commented on package.json:
```diff
@@ -1,66 +1,78 @@
+        "preview": "vite preview",
+        "prepare": "husky"
+    },
+    "lint-staged": {
```
Added ` Husky` and lint-staged configuration to enforce code quality standards before commits. This setup automatically:
- Runs ESLint with auto-fix on staged TypeScript files
- Formats all staged TS/TSX files with Prettier
- Formats staged JSON and Markdown files with Prettier

This pre-commit hook runs only on staged files before commit and automatically validates and formats code, ensuring consistent code style across the project without requiring manual checks before each commit.

Note: Manual `lint` and `format` scripts remain available for on-demand checks of the entire codebase when needed.

Having control over checking/formatting files without having to stage them, while automating the process for only the changed files, gives us more flexibility and freedom.

iamalaziz commented on src/router/index.tsx:
```diff
@@ -2,7 +2,7 @@
 
 import { Navigate, Route, Routes } from 'react-router-dom';
-import { routes } from './routes';
+import { routes } from '@router/routes';
```
*(medium)* Please fix the wrong import. Every change should be tested running the app and ensuring everything is working as expected.

<img width="742" alt="Screenshot 2025-04-26 at 12 27 57 AM" src="https://github.com/user-attachments/assets/e3af23fb-888c-427f-9264-08faec15b067" />

Sardor-M replied:
I see, sorry for this minor issue. it's already fixed.

iamalaziz reviewed (commented):
Could you resolve merge conflicts and fix the issues in the comments below. 
* There are still linting issues.
* The `husky` doesn't seem to be using pre-defined rules in the `.prettierrc`.

<img width="358" alt="Screenshot 2025-04-26 at 12 32 48 AM" src="https://github.com/user-attachments/assets/8b87a7d8-36ce-4168-89ce-b61b89e2b26d" />

Sardor-M commented:
> Could you resolve merge conflicts and fix the issues in the comments below.
> 
> * There are still linting issues.
> * The `husky` doesn't seem to be using pre-defined rules in the `.prettierrc`.
> 
> <img alt="Screenshot 2025-04-26 at 12 32 48 AM" width="358" src="https://private-user-images.githubusercontent.com/81867375/437511205-8b87a7d8-36ce-4168-89ce-b61b89e2b26d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU3NDQxNjMsIm5iZiI6MTc0NTc0Mzg2MywicGF0aCI6Ii84MTg2NzM3NS80Mzc1MTEyMDUtOGI4N2E3ZDgtMzZjZS00MTY4LTg5Y2UtYjYxYjg5ZTJiMjZkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI3VDA4NTEwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcwM2YyY2JmZWRmMjcxMmY2MGU5NzMzZWMxZmNiNjU2ODk2ODNkMTlmZjM3Yzc2NzgxYTcxOTA3ZGQ5OGVmYTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.BmxY1xEM-btC-Aa_JQd9kXA4MMIp_QsDBNfiILdylls">

These errors which are coming from the kakao type issues are expected as far i have mentioned before  these comments above, since giving the specific type instead of any would cause to break the kakao map drawing process, so that's is expected and on purpose !

You do not seems to read my comment in detail, as you may read it through, the husky will only touch on the files that are staged:  

> This pre-commit hook runs only on staged files before commit and automatically validates and formats code, ensuring consistent code style across the project without requiring manual checks before each commit.

i mentioned this part in the above comments !

And also pls make sure to check and test by yourself before writing it out if something is not working or not behaving as expected, you can refer to the attached photo.

Below as you can see the ``husky`` runned on the files that are changed and staged !
<img width="1243" alt="Screenshot 2025-04-27 at 5 52 33 PM" src="https://github.com/user-attachments/assets/f2b7b5a2-c645-4b31-a5e2-ae454daac920" />
